src/main/java/edu/kit/kastel/sdq/scorestats/cli/CLI.java

```diff
@@ -158,4 +159,32 @@ public void run(String[] args) {
             logger.info("Finished!");
         }
     }
+
+    private Assessments loadAssessments(GradingConfig config, ProgrammingExercise exercise) throws ArtemisNetworkException, AnnotationMappingException {
+        Collection<ProgrammingSubmission> submissions = new ArrayList<>(exercise.fetchSubmissions(0, false));
+
+        if (exercise.hasSecondCorrectionRound()) {
+            submissions.addAll(exercise.fetchSubmissions(1, false));
+        }
+
+        Map<String, Assessment> assessments = HashMap.newHashMap(submissions.size());
+        List<String> skippedStudents = new ArrayList<>();
+
+        for (ProgrammingSubmission submission : submissions) {
+            String studentId = submission.getParticipantIdentifier();
+            Assessment assessment = submission.openAssessment(config).orElse(null);
+            if (assessment == null) {
+                skippedStudents.add(studentId);
+                continue;
+            }
+
+            if (assessments.containsKey(studentId)) {
```

Comment:
If you load two correction rounds I guess this breaks.

Reply:
It seemed to work in the old code: https://github.com/kit-sdq/programming-lecture-artemis-score-stats/blob/c7f6781b8b436e1d3cc526986d70f09e591f322e/src/main/java/edu/kit/kastel/sdq/scorestats/core/client/Artemis4JArtemisClient.java#L94

Reply:
Ok .. then. Did you tested the code for the test system or on production for your students?

Reply:
I only tested it in production

Reply:
Perfect. I'll test it as well tomorrow :)

Reply:
Especially Ill test it with exams .. they have two correction rounds :D